sessionctx/binloginfo/binloginfo.go

```diff
@@ -55,6 +58,40 @@ func GetPumpsClient() *pumpcli.PumpsClient {
 	return client
 }
 
+// GetOrCreatePumpsClient get or create the pumps client instance
+func GetOrCreatePumpsClient() *pumpcli.PumpsClient {
+	pumpsClientLock.Lock()
```

Comment:
we can use read lock here to get the client, if client is nil, then lock and create client.

Reply:
Done.